raiden-ts/src/channels/epics.ts

```diff
@@ -61,6 +66,54 @@ import {
   channelWithdrawn,
 } from './actions';
 
+/**
+ * Receives an async function and returns an observable which will retry it every interval until it
+ * resolves, or throw if it can't succeed after 10 retries.
+ * It is needed e.g. on provider methods which perform RPC requests directly, as they can fail
+ * temporarily due to network errors, so they need to be retried for a while.
+ * JsonRpcProvider._doPoll also catches, suppresses & retry
+ *
+ * @param func - An async function (e.g. a Promise factory, like a defer callback)
+ * @param interval - Interval to retry in case of rejection
+ * @param retries - Max number of times to retry
+ * @returns Observable version of async function, with retries
+ */
+function retryAsync$<T>(func: () => Promise<T>, interval = 1e3, retries = 10): Observable<T> {
+  return defer(func).pipe(
+    retryWhen(err$ =>
+      err$.pipe(mergeMap((err, i) => (i < retries ? timer(interval) : throwError(err)))),
+    ),
+  );
+}
+
+/**
+ * Custom operator to wait & assert transaction success
+ *
+ * @param method - method name to use in logs
+ * @param error - ErrorCode to throw if transaction fails
+ * @param deps - object containing logger
+ * @returns operator function to wait for transaction and output hash
+ */
+function assertTx(
```

Comment:
Nice addition :D

Reply:
Thanks! It's staged to be moved to channels utils, so I can use it on the SecretRegistry epics on transfers as well! but no logic change